server/src/routes/api/v1/dialogsRouter.js

```diff
@@ -0,0 +1,77 @@
+import express from "express"
+import { ValidationError } from "objection"
+import { Dialog, User } from "../../../models/index.js"
+import { dialogsForList } from "./serializers/dialogsSerializer.js"
+
+const dialogsRouter = new express.Router()
+
+dialogsRouter.get("/", async (req, res) => {
+    try {
+        if (!req.user) {
+            return (
+                res
+                .set({"Content-Type": "application/json"})
+                .status(401)
+                .json({ errors: "must be logged in" })
+            )
+        } 
+        
+        const requestingUser = parseInt(req.user.id)
+        const userToRetrieve = req.query.user_id ? parseInt(req.query.user_id) : requestingUser
```

Comment:
not sure why `parseInt` is needed? `findById` can accept a string integer as an argument